src/coinjoin/client.h

```diff
@@ -72,10 +70,50 @@ class CPendingDsaRequest
     }
 };
 
+struct CJClientManager {
+    CJClientManager(CConnman& connman, CTxMemPool& mempool, const CMasternodeSync& mn_sync,
+                    const std::unique_ptr<CCoinJoinClientQueueManager>& queueman)
+        : m_connman(connman), m_mempool(mempool), m_mn_sync(mn_sync), m_queueman(queueman) {}
+    ~CJClientManager() {
+        for (auto& pair : m_imap) {
+            pair.second.reset();
+        }
+    }
+
+    void Add(CWallet& wallet);
+    void DoMaintenance(CBlockPolicyEstimator& fee_estimator);
+
+    void Remove(const std::string& name) {
+        auto it = m_imap.find(name);
+        if (it != m_imap.end()) { m_imap.erase(it); }
+    }
+
+    CCoinJoinClientManager* Get(const CWallet& wallet) const {
+        auto it = m_imap.find(wallet.GetName());
+        return (it != m_imap.end()) ? it->second.get() : nullptr;
+    }
+
+    std::map<const std::string, std::unique_ptr<CCoinJoinClientManager>>& raw() { return m_imap; }
+
+private:
+    CConnman& m_connman;
+    CTxMemPool& m_mempool;
+
+    const CMasternodeSync& m_mn_sync;
+    const std::unique_ptr<CCoinJoinClientQueueManager>& m_queueman;
+
+    std::map<const std::string, std::unique_ptr<CCoinJoinClientManager>> m_imap;
```

Comment:
m_imap is not very descriptive; something like `m_wallet_name_manager_map` I know its long but at least its descriptive

Reply:
Renamed to `m_wallet_manager_map`. Resolved in latest push.